extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/DefaultChunkReadingFactory.java

```diff
@@ -184,19 +166,131 @@ public ChunkReader getReader(StreamReaderOptions options, int factor,
+                    BigInteger::new,
+                    outChunk,
+                    outOffset,
+                    totalRows);
+        }
+    }
+
+    private enum BigDecimalChunkReader implements ChunkReader {
+        BIG_DECIMAL_CHUNK_READER;
+
+        @Override
+        public WritableChunk<Values> readChunk(Iterator<ChunkInputStreamGenerator.FieldNodeInfo> fieldNodeIter,
+                PrimitiveIterator.OfLong bufferInfoIter, DataInput is, WritableChunk<Values> outChunk, int outOffset,
+                int totalRows) throws IOException {
+            return VarBinaryChunkInputStreamGenerator.extractChunkFromInputStream(
+                    is,
+                    fieldNodeIter,
+                    bufferInfoIter,
+                    BigDecimalChunkReader::readBigDecimal,
+                    outChunk,
+                    outOffset,
+                    totalRows);
+        }
+
+        private static BigDecimal readBigDecimal(byte[] buf, int offset, int length) {
+            // read the int scale value as little endian, arrow's endianness.
+            final byte b1 = buf[offset];
+            final byte b2 = buf[offset + 1];
+            final byte b3 = buf[offset + 2];
+            final byte b4 = buf[offset + 3];
+            final int scale = b4 << 24 | (b3 & 0xFF) << 16 | (b2 & 0xFF) << 8 | (b1 & 0xFF);
+            return new BigDecimal(new BigInteger(buf, offset + 4, length - 4), scale);
+        }
+    }
+
+    private enum StringChunkReader implements ChunkReader {
+        STRING_CHUNK_READER;
+
+        @Override
+        public WritableChunk<Values> readChunk(Iterator<ChunkInputStreamGenerator.FieldNodeInfo> fieldNodeIter,
+                PrimitiveIterator.OfLong bufferInfoIter, DataInput is, WritableChunk<Values> outChunk, int outOffset,
+                int totalRows) throws IOException {
+            return VarBinaryChunkInputStreamGenerator.extractChunkFromInputStream(
+                    is,
+                    fieldNodeIter,
+                    bufferInfoIter,
+                    StringChunkReader::readString,
+                    outChunk,
+                    outOffset,
+                    totalRows);
+        }
+
+        private static String readString(byte[] buf, int off, int len) {
+            return new String(buf, off, len, Charsets.UTF_8);
+        }
+    }
+
+    private enum SchemaChunkReader implements ChunkReader {
+        SCHEMA_CHUNK_READER;
+
+        @Override
+        public WritableChunk<Values> readChunk(Iterator<ChunkInputStreamGenerator.FieldNodeInfo> fieldNodeIter,
+                PrimitiveIterator.OfLong bufferInfoIter, DataInput is, WritableChunk<Values> outChunk, int outOffset,
+                int totalRows) throws IOException {
+            return VarBinaryChunkInputStreamGenerator.extractChunkFromInputStream(
+                    is,
+                    fieldNodeIter,
+                    bufferInfoIter,
+                    ArrowIpcUtil::deserialize,
```

Comment:
I'm not sure why this is showing up in the diff; this has already been merged to main. It may be due to merge conflicts I had to resolve, or something like that... let me merge again to get up to date wrt to main...